Realm/RealmSwift/Realm.swift

```diff
@@ -58,15 +58,31 @@ public class Realm {
     }
 
     public convenience init(path: String) {
-        self.init(path: path, readOnly: false, error: nil)
+        self.init(rlmRealm: RLMRealm(path: path, readOnly: false, error: nil))
     }
 
     public convenience init(inMemoryIdentifier: String) {
         self.init(rlmRealm: RLMRealm.inMemoryRealmWithIdentifier(inMemoryIdentifier))
     }
 
-    public convenience init(path: String, readOnly readonly: Bool, error: NSErrorPointer) {
-        self.init(rlmRealm: RLMRealm(path: path, readOnly: readonly, error: error))
+    public convenience init?(path: String, readOnly readonly: Bool, error: NSErrorPointer) {
+        if let rlmRealm = RLMRealm(path: path, readOnly: readonly, error: error) as RLMRealm? {
+            self.init(rlmRealm: rlmRealm)
+        }
+        else {
+            self.init(rlmRealm: RLMRealm.inMemoryRealmWithIdentifier(NSUUID().UUIDString))
```

Comment:
Obviously we can't return nil in the no-op function I mentioned earlier, since that gets us right back to where we were before, but we should still find a minimal way to please the compiler, without having to actually create a legitimate RLMRealm.